lib/logger/lib/logger/legacy_handler.ex

```diff
@@ -0,0 +1,265 @@
+defmodule Logger.LegacyHandler do
+  @moduledoc false
+
+  alias Logger.Counter
+
+  @internal_keys [:counter]
+
+  @doc false
+  def child_spec(start_options) do
+    %{
+      id: :gen_event,
+      start: {:gen_event, :start_link, [{:local, Logger}, start_options]},
+      modules: :dynamic
+    }
+  end
+
+  def adding_handler(%{config: data} = config) do
+    # TODO: When using counters exclusively then this line can be changed to use
+    # `Map.put/2` instead as we will not need to pass ETS table
+    new_data =
+      Logger.Config.default_config()
+      |> Map.merge(data || %{})
+      |> Map.put_new_lazy(:counter, &Logger.Config.new/0)
```

Comment:
So when we ask the handler to be added on Logger.App.start, we pass the counter. Why do we need to pass it here again? Is it because someone may add it *directly*? If so, I don't care about this use case. We can raise if someone tries to add it directly, especially because it won't work without the Logger process anyway.

Reply:
It is just in case when someone will use `:logger.set_handler_config/2` by the accident instead of `:logger.update_handler_config/2`.

Reply:
This shouldn't impact adding handler, right?

Reply:
No it should not.